File: app/models/national_governing_body.rb
# == Schema Information
#
# Table name: national_governing_bodies
#
#  id           :bigint(8)        not null, primary key
#  acronym      :string
#  country      :string
#  image_url    :string
#  name         :string           not null
#  player_count :integer          default(0), not null
#  region       :integer
#  website      :string
#  created_at   :datetime         not null
#  updated_at   :datetime         not null
#
# Indexes
#
#  index_national_governing_bodies_on_region  (region)
#

# Data model for NationalGoverningBodies, currently only used for Referees to assign for themselves and users to search
# based this association
class NationalGoverningBody < ApplicationRecord
  require 'activerecord-import/base'
  require 'activerecord-import/active_record/adapters/postgresql_adapter'

  enum region: {
    north_america: 0,
    south_america: 1,
    europe: 2,
    africa: 3,
    asia: 4
  }

  has_many :referee_locations, dependent: :destroy
  has_many :referees, through: :referee_locations
  has_many :certified_referees, -> { certified }, through: :referee_locations, source: :referee
  has_many :teams, dependent: :destroy
  has_many :stats, inverse_of: :national_governing_body, class_name: 'NationalGoverningBodyStat', dependent: :destroy
  has_many :social_accounts, as: :ownable, dependent: :destroy
end

File: dotnet/ManagementHub.Models/NationalGoverningBody.cs
using System;
using System.Collections.Generic;

namespace ManagementHub.Models
{
    public partial class NationalGoverningBody
    {
        public NationalGoverningBody()
        {
            NationalGoverningBodyAdmins = new HashSet<NationalGoverningBodyAdmin>();
            NationalGoverningBodyStats = new HashSet<NationalGoverningBodyStat>();
            RefereeLocations = new HashSet<RefereeLocation>();
            Teams = new HashSet<Team>();
        }

        public long Id { get; set; }
        public string Name { get; set; } = null!;
        public string? Website { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int PlayerCount { get; set; }
        public string? ImageUrl { get; set; }
        public string? Country { get; set; }
        public string? Acronym { get; set; }
        public int? Region { get; set; }
        public int MembershipStatus { get; set; }

        public virtual ICollection<NationalGoverningBodyAdmin> NationalGoverningBodyAdmins { get; set; }
        public virtual ICollection<NationalGoverningBodyStat> NationalGoverningBodyStats { get; set; }
        public virtual ICollection<RefereeLocation> RefereeLocations { get; set; }
        public virtual ICollection<Team> Teams { get; set; }
    }
}
